RestaurantReservation.API/Controllers/ReservationsController.cs

```diff
@@ -0,0 +1,208 @@
+        [HttpPut("{id}")]
+        public async Task<ActionResult> Update(int id, ReservationDto reservation)
+        {
+            var existingReservation = await _reservationRepository.GetByIdAsync(id);
+            if (existingReservation == null)
+            {
+                var error = ReservationErrors.NotFound(id);
+                return NotFound(error);
+            }
+
+            var matchingRestaurantTable = await _reservationRepository.MatchingTableRestaurant(reservation.RestaurantId, reservation.TableId);
+            if (!matchingRestaurantTable)
+            {
+                var error = ReservationErrors.NotMatchingTableRestaurant(reservation.RestaurantId, reservation.TableId);
+                return NotFound(error);
+            }
+
+            _mapper.Map(reservation, existingReservation);
+            await _reservationRepository.UpdateAsync(existingReservation);
+            return NoContent();
+        }
+
+        /// <summary>
+        /// Get reservations for a specific customer
+        /// </summary>
+        /// <param name="customerId">Id of the customer</param>
+        /// <returns>List of reservations for the customer</returns>
+        /// <response code="200">Returns the list of reservations for the customer</response>
+        /// <response code="404">If no reservations found for the customer</response>
+        [HttpGet("customer/{customerId}")]
+        public async Task<ActionResult<IEnumerable<ReservationWithoutDetailsDto>>> GetReservationForCustomer(int customerId)
+        {
+            var reservation = await _reservationRepository.GetReservationsByCustomerAsync(customerId);
+            if (reservation.Count() == 0)
+            {
+                var error = ReservationErrors.NoReservationFound(customerId);
+                return NotFound(error);
+            }
+            return Ok(_mapper.Map<IEnumerable<ReservationWithoutDetailsDto>>(reservation));
+        }
+
+        /// <summary>
+        /// Get orders with related menu items for a specific reservation
+        /// </summary>
+        /// <param name="reservationId">Id of the reservation</param>
+        /// <returns>List of orders with related menu items</returns>
+        /// <response code="200">Returns the list of orders with related menu items for the reservation</response>
+        /// <response code="404">If no reservation with the specified id is found</response>
+        [HttpGet("{reservationId}/orders")]
+        public async Task<ActionResult<IEnumerable<OrderWithOrdersItemsDto>>> GetReservationOrdersWithRelatedMenuItems(int reservationId)
+        {
+            var reservation = await _reservationRepository.GetByIdAsync(reservationId);
+            if (reservation == null)
+            {
+                var error = ReservationErrors.NotFound(reservationId);
+                return NotFound(error);
+            }
+            var details = await _reservationRepository.ListOrdersAndMenuItemsAsync(reservationId);
```

Comment:
List be consistence in naming our repository methods, Retrieve/Get - don't really like List I'd explain.